cider/type/plan/Analyzer.h

```diff
@@ -1940,6 +1940,48 @@ class SubstringStringOper : public StringOper {
   }
 };
 
+class ConcatStringOper : public StringOper {
+ public:
+  ConcatStringOper(const std::shared_ptr<Analyzer::Expr>& former,
+                   const std::shared_ptr<Analyzer::Expr>& latter)
+      : StringOper(getConcatOpKind({former, latter}),
+                   rearrangeOperands({former, latter}),
+                   getMinArgs(),
+                   getExpectedTypeFamilies(),
+                   getArgNames()) {}
+
+  ConcatStringOper(const std::vector<std::shared_ptr<Analyzer::Expr>>& operands)
+      : StringOper(getConcatOpKind(operands),
+                   rearrangeOperands(operands),
+                   getMinArgs(),
+                   getExpectedTypeFamilies(),
+                   getArgNames()) {}
+
+  ConcatStringOper(const std::shared_ptr<Analyzer::StringOper>& string_oper)
+      : StringOper(string_oper) {}
+
+  std::shared_ptr<Analyzer::Expr> deep_copy() const override;
+
+  size_t getMinArgs() const override { return 2UL; }
+
+  std::vector<OperandTypeFamily> getExpectedTypeFamilies() const override {
+    return {OperandTypeFamily::STRING_FAMILY, OperandTypeFamily::STRING_FAMILY};
+  }
+
+  std::vector<std::string> getArgNames() const override {
```

Comment:
```c++
const std::vector<std::string>& getArgNames() const override {
  static std::vector<std::string> names = {"operand_0", "operand_1"};
  return names;
}
```
maybe you should change the function prototype in base class.

Reply:
done